src/qz/utils/SystemUtilities.java

```diff
@@ -156,30 +156,22 @@ public static Version getJavaVersion(String version) {
 
     /**
      * Determines the currently running Jar's absolute path on the local filesystem
+     * todo: make this return a sane directory for running via ide
      *
      * @return A String value representing the absolute path to the currently running
      * jar
      */
-    public static String detectJarPath() {
+    public static String getJarPath() {
+        if (jarPath != null) {
```

Comment:
This is redundant, just guard the whole statement, always return the same value.